Library/Homebrew/rubocops/lines.rb

```diff
@@ -517,14 +524,64 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
 
         # matches ({bash,zsh,fish}_completion/"_?foo{.fish}?").write output
         def_node_search :shell_completion_node, <<~EOS
-        $(send
-          (begin
-            (send
-              (send nil? {:bash_completion :zsh_completion :fish_completion}) :/
-              (str _))) :write _)
+          $(send
+            (begin
+              (send
+                (send nil? {:bash_completion :zsh_completion :fish_completion}) :/
+                (str _))) :write _)
         EOS
       end
 
+      # This cop makes sure that the `generate_completions_from_executable` DSL is used with only
+      # a single, combined call for all shells.
+      #
+      # @api private
+      class SingleGenerateCompletionsDSLCall < FormulaCop
+        extend AutoCorrector
+
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          install = find_method_def(body_node, :install)
+
+          methods = find_every_method_call_by_name(install, :generate_completions_from_executable)
+          return if methods.length <= 1
+
+          offenses = []
+          shells = []
+          methods.each do |method|
+            next unless method.source.include?("shells:")
+
+            shells << method.source.match(/shells: \[(:bash|:zsh|:fish)\]/).captures.first
+            offenses << method
+          end
+
+          return if offenses.blank?
+
+          offenses[0...-1].each do |node|
+            offending_node(node)
+            problem "Use a single `generate_completions_from_executable` call
+                         combining all specified shells" do |corrector|
+              corrector.replace(@offensive_node.source_range, "")
```

Comment:
Not sure this will work, but how about extending the replacement range by 1 extra character to include the `\n`, and then replacing that all with `""`?